packages/core-mobile/app/services/notifications/balanceChange/registerDeviceToNotificationSender.ts

```diff
@@ -0,0 +1,32 @@
+import Logger from 'utils/Logger'
+import AppCheckService from 'services/fcm/AppCheckService'
+
+export async function registerDeviceToNotificationSender(
+  fcmToken: string
+): Promise<{ deviceArn: string }> {
+  const appCheckToken = await AppCheckService.getToken()
+  const options: RequestInit = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'X-Firebase-AppCheck': appCheckToken.token
+    },
+    body: JSON.stringify({
+      deviceToken: fcmToken,
+      appType: 'CORE_MOBILE_APP'
+    })
```

Comment:
will be great to extract this logic to a func `fetchWithAppcheck` or `coreFetch` and we can reuse the fetch function

Reply:
something like this?
[Extract AppCheckService.fetch function](https://github.com/ava-labs/core-mobile/pull/1659/commits/53341134ef1900763c0837744e1aa635d2b08adc)